src/workloads/query/LookupColocatedData.yml

```diff
@@ -0,0 +1,368 @@
+      MetricsName: ShardedLocalFieldForeignField1k
+      Coll: *ShardedLocal
+      BatchSize: *BatchSize1k
+
+- ActorFromTemplate:
+    TemplateName: SelfLookupLocalFieldForeignField
+    TemplateParameters:
+      OnlyActiveInPhase: 13
+      MetricsName: ShardedLocalFieldForeignField5k
+      Coll: *ShardedLocal
+      BatchSize: *BatchSize5k
+
+- ActorFromTemplate:
+    TemplateName: SelfLookupLocalFieldForeignField
+    TemplateParameters:
+      OnlyActiveInPhase: 14
+      MetricsName: ShardedLocalFieldForeignField15k
+      Coll: *ShardedLocal
+      BatchSize: *BatchSize15k
+
+- ActorFromTemplate:
+    TemplateName: SelfLookupLetPipeline
+    TemplateParameters:
+      OnlyActiveInPhase: 15
+      MetricsName: ShardedLetPipeline1k
+      Coll: *ShardedLocal
+      BatchSize: *BatchSize1k
+
+- ActorFromTemplate:
+    TemplateName: SelfLookupLetPipeline
+    TemplateParameters:
+      OnlyActiveInPhase: 16
+      MetricsName: ShardedLetPipeline5k
+      Coll: *ShardedLocal
+      BatchSize: *BatchSize5k
+
+- ActorFromTemplate:
+    TemplateName: SelfLookupLetPipeline
+    TemplateParameters:
+      OnlyActiveInPhase: 17
+      MetricsName: ShardedLetPipeline15k
+      Coll: *ShardedLocal
+      BatchSize: *BatchSize15k
+
+#
+# Sharded, all on same shard
+#
+- ActorFromTemplate:
+    TemplateName: SelfLookupLocalFieldForeignField
+    TemplateParameters:
+      OnlyActiveInPhase: 18
+      MetricsName: ShardedAllOnOneShardLocalFieldForeignField1k
+      Coll: *ShardedLocalAllOnOneShard
+      BatchSize: *BatchSize1k
+
+- ActorFromTemplate:
+    TemplateName: SelfLookupLocalFieldForeignField
+    TemplateParameters:
+      OnlyActiveInPhase: 19
+      MetricsName: ShardedAllOnOneShardLocalFieldForeignField5k
+      Coll: *ShardedLocalAllOnOneShard
+      BatchSize: *BatchSize5k
+
+- ActorFromTemplate:
+    TemplateName: SelfLookupLocalFieldForeignField
+    TemplateParameters:
+      OnlyActiveInPhase: 20
+      MetricsName: ShardedAllOnOneShardLocalFieldForeignField15k
+      Coll: *ShardedLocalAllOnOneShard
+      BatchSize: *BatchSize15k
+
+- ActorFromTemplate:
+    TemplateName: SelfLookupLetPipeline
+    TemplateParameters:
+      OnlyActiveInPhase: 21
+      MetricsName: ShardedAllOnOneShardLetPipeline1k
+      Coll: *ShardedLocalAllOnOneShard
+      BatchSize: *BatchSize1k
+
+- ActorFromTemplate:
+    TemplateName: SelfLookupLetPipeline
+    TemplateParameters:
+      OnlyActiveInPhase: 22
+      MetricsName: ShardedAllOnOneShardLetPipeline5k
+      Coll: *ShardedLocalAllOnOneShard
+      BatchSize: *BatchSize5k
+
+- ActorFromTemplate:
+    TemplateName: SelfLookupLetPipeline
+    TemplateParameters:
+      OnlyActiveInPhase: 23
+      MetricsName: ShardedAllOnOneShardLetPipeline15k
+      Coll: *ShardedLocalAllOnOneShard
+      BatchSize: *BatchSize15k
+
+AutoRun:
+- When:
+    branch_name:
+      $neq:
+      - v4.0
+      - v4.2
+      - v4.4
+      - v5.0
+    mongodb_setup:
+      $eq:
+      - shard
```

Comment:
It may not be worth listing all of these configurations here, maybe we only need one or two? This first config is for the 3-shard cluster, while the last two are for 2-shard clusters. And as expected, $lookup is faster in the 3-shard case when the data is distributed across all 3 shards. Not really sure if this gives us useful intel. 

One thing to note is that if we introduce a sharded force-classic variant, we will need to pick which of these configs to mimic-- 3-shard or 2-shard? Feature flags on or off?

Reply:
I would probably schedule the test on the following variants:
* Linux Shard Lite Cluster 2022-11
* Linux 3-Shard Cluster 2022-11
* Linux Shard Lite Cluster Intel 2022-11
* And maybe Linux Shard Lite (all feature flags) 2022-11 if the behavior under test here could be impacted by feature flags

In case you decide to also add a new variant with forceClassic, I would probably stick to 1-2 variants.